### PR TITLE
[BugFix] Make DatabaseTransactionMgr.getMinActiveTxnId() thread safe

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java [hunk 1]
```diff
@@ -107,11 +107,13 @@ private boolean allCommittedTransactionsBeforeRestartHaveFinished() {
...
         }
+        // Note: must call getMinActiveTxnId() before getNextTransactionId(), otherwise if there are no running transactions
+        // waitTxnId <= minActiveTxnId will always be false.
+        long minActiveTxnId = transactionMgr.getMinActiveTxnId();
         if (waitTxnId < 0) {
             waitTxnId = transactionMgr.getTransactionIDGenerator().getNextTransactionId();
         }
-        Long minActiveTxnId = transactionMgr.getMinActiveTxnId();
-        finishedWaiting = (minActiveTxnId == null) || minActiveTxnId > waitTxnId;
+        finishedWaiting = waitTxnId <= minActiveTxnId;
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java:116]: `return waitTxnId <= minActiveTxnId`

**Review by kevincai — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 19 (78.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionScheduler.java|0|2|00.00%|[112, 116]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|0|1|00.00%|[918]|
|:large_blue_circle:|com/starrocks/alter/RollupJobV2.java|0|1|00.00%|[871]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by sduzh** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**